### PR TITLE
Fix special protocols that indexes are not correctly retrieved

### PR DESCRIPTION
**Pull request by chuan-wang**

### CODE REVIEW

File: LIMS2DB/classes.py [hunk 1]
```diff
@@ -770,10 +770,16 @@ def get_library_preps(self, sample):
...
-                        if 'By user' not in self.obj['details']['library_construction_method'] and inp_artifact.reagentlabels:
+                        if 'By user' not in self.obj['details']['library_construction_method'] and len(inp_artifact.reagentlabels)==1:
                             # if finlib, these are already computed
                             self.obj['samples'][sample.name]['library_prep'][prepname]['reagent_label'] = inp_artifact.reagentlabels[0].name
                             self.obj['samples'][sample.name]['library_prep'][prepname]['barcode'] = self.extract_barcode(inp_artifact.reagentlabels[0].name)
+                        elif 'By user' not in self.obj['details']['library_construction_method'] and len(inp_artifact.reagentlabels)>1:
+                            # For cases that samples are indexed and pooled prior to Library QC
+                            for iaa in inp_artifact.ancestors:
+                                if iaa.reagentlabels and len(iaa.samples)==1 and iaa.samples[0].name==sample.name:
```
Comment (alneberg) [inline LIMS2DB/classes.py]: Do you need an else-case for this or is it fine to just ignore other cases?

Comment (chuan-wang) [reply LIMS2DB/classes.py]: The else case is for finished libraries, and it has been handled in Line 712-723

Comment (alneberg) [reply LIMS2DB/classes.py]: I meant the `if` on line 780. Was that the one you answered on as well, or was it the one on line `777`?

Comment (chuan-wang) [reply LIMS2DB/classes.py]: The `else` cases of line 780 are irrelevant artefacts which can be ignored.

Comment (alneberg) [reply LIMS2DB/classes.py]: Ok, that makes sense! I'm not really comfortable with this codebase yet...

**Comment by chuan-wang** [comment]
Thanks!

**Pull request merged**